src/data/java_profile.rs

```diff
@@ -0,0 +1,340 @@
+                .insert(pid.clone(), vec![String::from("Could not resolve name!")]);
+        }
+
+        let jps_str = self.update_process_map();
+        let jps: Vec<&str> = jps_str.split_whitespace().collect();
+
+        let default = String::from("jps");
+        let jprofile = params
+            .profile
+            .get(JAVA_PROFILE_FILE_NAME)
+            .unwrap_or(&default)
+            .as_str();
+        match jprofile {
+            "jps" => {
+                jids = self.process_map.clone().into_keys().collect();
+                debug!("Jps will be run if new JVM is started during aperf record to resolve process names.",);
+            }
+            _ => {
+                let args: Vec<&str> = jprofile.split(',').collect();
+                for arg in args {
+                    if !jps.contains(&arg) {
+                        error!("No JVM with name/PID '{}'.", arg);
+                        continue;
+                    } else if jps.iter().position(|&r| r == arg).unwrap()
+                        != jps.iter().rposition(|&r| r == arg).unwrap()
+                    {
+                        error!("Multiple JVMs with the name '{}', please provide PID.", arg);
+                        continue;
+                    }
+                    if let Some(jid) = self.get_jid(arg) {
+                        jids.push(jid);
+                    }
+                }
+            }
+        }
+
+        self.launch_asprof(jids, params)
+    }
+
+    fn collect_data(&mut self, params: CollectorParams) -> Result<()> {
+        let jprofile = params.profile.get(JAVA_PROFILE_FILE_NAME).unwrap().as_str();
+        if jprofile != "jps" {
+            return Ok(());
+        }
+
+        let pgrep_out = Command::new("pgrep").arg("java").output().unwrap();
+        let pgrep_str = String::from_utf8(pgrep_out.stdout).unwrap();
+        let pgrep: Vec<&str> = pgrep_str.split_whitespace().collect();
+
+        let mut jids: Vec<String> = Vec::new();
+        for pid in pgrep {
+            if self.process_map.contains_key(pid) {
+                continue;
+            }
+            self.process_map.insert(
+                String::from(pid),
+                vec![String::from("Could not resolve name!")],
+            );
+            jids.push(String::from(pid));
+        }
+
+        if jids.is_empty() || params.elapsed_time >= params.collection_time {
+            return Ok(());
+        }
+
+        self.update_process_map();
+        self.launch_asprof(jids, params)
+    }
+
+    fn finish_data_collection(&mut self, params: CollectorParams) -> Result<()> {
+        trace!("Waiting for asprof profile collection to complete...");
+        while ASPROF_CHILDREN.lock().unwrap().len() > 0 {
+            match ASPROF_CHILDREN.lock().unwrap().pop().unwrap().wait() {
+                Err(e) => {
+                    error!("'asprof' did not exit successfully: {}", e);
+                    return Ok(());
+                }
```

Comment:
This happens if process is owned by another user, right?

Reply:
I think so, and any other error when async-profiler is started properly.

Reply:
It is inherited from aperf process.